### DISCUSSION TITLE
Unable to connect when deployed to Heroku

### DISCUSSION THREAD

**Discussion by Ethan7** [geckosio/geckos.io#108] [Help / Questions] [answered]
When I try to deploy my app which implements geckosio to Heroku I get this error:
"POST https://url-here/:19486/.wrtc/v1/connections 404 (Not Found)"
To be clear the port is correct, it is set by heroku and then I have it passed to the clientside via a WebSocket which finds the correct port. Then I pass that port to the geckosio initialization. The only similar error I'd seen in the geckosio issues list was solved by replacing the run code with "npm run production", so I tried this and changed the package.json to match, and nothing changed. I still don't know why I'm unable to open a geckosio connection to my heroku app using nodejs. Any help is appreciated.
Here's the involved Server-side code:
```
const express = require('express');
const app = express();
const server = require('http').Server(app);
const geckos = require('@geckos.io/server').default;
const geckosio = geckos();
geckosio.addServer(server);
const PORT = process.env.PORT || 8080;
server.listen(PORT, function(){
	console.log("Server is now running on port "+PORT);
});
```

And here's the involved client-side code:
```
Socket.on('socketID', function(data){
  port = data.port;
  
  //geckos io channel
  const channel = geckos({url:'insert url here', port:port});
});
```

**Comment by yandeu** [comment, marked as answer]
See https://github.com/geckosio/geckos.io/issues/71#issuecomment-748620234

**Reply by Ethan7** [reply]
Do you think Heroku will start allowing UDP connections when they upgrade their system to QUIC and HTTP/3?

**Reply by Ethan7** [reply]
I may have to rescind my accepted answer there, I'm finding WebRTC projects on Heroku like this https://elements.heroku.com/buttons/renanpupin/nodejs-webrtc-video-stream and this https://elements.heroku.com/buttons/sandrolaudares/symple-webrtc-video-chat-demo and this https://elements.heroku.com/buttons/singhmukesh/webrtc and I don't see how these would be possible without UDP. I'm going to be looking into this further.

**Reply by yandeu** [reply]
> Do you think Heroku will start allowing UDP connections when they upgrade their system to QUIC and HTTP/3?

Upgrading to QUIC or HTTP/3 has no effect on supporting forwarding public user-defined UDP ports to your dynos.

> WebRTC projects on Heroku

These projects use the "normal" WebRTC use case. Normally 2 clients (browsers) do connect directly with each other via UDP. There, you only need a signaling server (using HTTP, WebSocket etc.) that will run everywhere.

But if you use geckos.io, the WebRTC traffic will go through the server. So browser => server => browser. That is the whole point of geckos.io: Using UDP on the server and not client-side only.

Hope it helps :)

**Reply by Ethan7** [reply]
I see! Thank you for the explaination. I remember looking up the difference between client-client and server-client webrtc when I first found geckosio, but since then I hadn't thought much of it.
